boot/bootstate20.go

```diff
@@ -456,7 +569,7 @@ func selectSuccessfulBootSnap(b bootState, update bootStateUpdate) (
 	// kernel_status and base_status go from "" -> "try" (set by snapd), to
 	// "try" -> "trying" (set by the boot script)
 	// so if we are in "trying" mode, then we should choose the try snap
-	if status == TryingStatus {
+	if status == TryingStatus && trySnap != nil {
```

Comment:
was this a bug? do we need a test for it?

Reply:
yes, the bug was that we were not disabling a try kernel when we should have been as a cleanup operation; the specific test that is affected is TestMarkBootSuccessful20KernelStatusTryingNoKernelSnapCleansUp, where we don't have a try-kernel on the bootloader, so TryKernel() returns nil, but the status is TryingStatus, so here we don't want to return the trySnap since it's nil, and it being nil prevents the cleanup code from running in bootState20MarkSuccessful.commit's method. 

If you want I could separate out this fix into a separate PR where it's simpler to reason about. I had meant to at least put this change into it's own commit but forgot about it when I went to push this up.

Reply:
to be clear, the test also was wrong in it's tests, there should have been DisableTryKernel calls, but I'm not sure why I wrote it such that there weren't any

Reply:
this change was broken out as https://github.com/snapcore/snapd/pull/8503